tests/estimators/certification/test_smooth_vit.py

```diff
@@ -0,0 +1,745 @@
+
+        # check that the ablation wraps when on the edge of the image
+        ablated = col_ablator.forward(cifar_data, column_pos=30)
+
+        assert ablated.shape[1] == 4
+        assert torch.sum(ablated[:, :, 30 * 7 :, :]) > 0
+        assert torch.sum(ablated[:, :, 2 * 7 : 30 * 7, :]) == 0
+        assert torch.sum(ablated[:, :, : 2 * 7, :]) > 0
+
+    except ARTTestException as e:
+        art_warning(e)
+
+
+@pytest.mark.only_with_platform("pytorch")
+def test_pytorch_training(art_warning, fix_get_mnist_data, fix_get_cifar10_data):
+    """
+    Check that the training loop for pytorch does not result in errors
+    """
+    import torch
+    from art.estimators.certification.derandomized_smoothing import PyTorchDeRandomizedSmoothing
+
+    try:
+        cifar_data = fix_get_cifar10_data[0][:50]
+        cifar_labels = fix_get_cifar10_data[1][:50]
+
+        art_model = PyTorchDeRandomizedSmoothing(
+            model="vit_small_patch16_224",
+            loss=torch.nn.CrossEntropyLoss(),
+            optimizer=torch.optim.SGD,
+            optimizer_params={"lr": 0.01},
+            input_shape=(3, 32, 32),
+            nb_classes=10,
+            ablation_size=4,
+            load_pretrained=True,
+            replace_last_layer=True,
+            verbose=False,
+        )
+
+        scheduler = torch.optim.lr_scheduler.MultiStepLR(art_model.optimizer, milestones=[1], gamma=0.1)
+        art_model.fit(cifar_data, cifar_labels, nb_epochs=2, update_batchnorm=True, scheduler=scheduler)
+
+    except ARTTestException as e:
+        art_warning(e)
+
+
+@pytest.mark.only_with_platform("pytorch")
+def test_certification_function(art_warning, fix_get_mnist_data, fix_get_cifar10_data):
+    """
+    Check that based on a given set of synthetic class predictions the certification gives the expected results.
+    """
+    from art.estimators.certification.derandomized_smoothing.derandomized_smoothing_pytorch import ColumnAblator
+    import torch
+
+    try:
+        col_ablator = ColumnAblator(
+            ablation_size=4,
+            channels_first=True,
+            mode="ViT",
+            to_reshape=True,  # do not upsample initially
+            original_shape=(3, 32, 32),
+            output_shape=(3, 224, 224),
+        )
+        pred_counts = torch.from_numpy(np.asarray([[20, 5, 1], [10, 5, 1], [1, 16, 1]]))
+        cert, cert_and_correct, top_predicted_class = col_ablator.certify(
+            pred_counts=pred_counts,
+            size_to_certify=4,
+            label=0,
+        )
+        assert torch.equal(cert, torch.tensor([True, False, True]))
+        assert torch.equal(cert_and_correct, torch.tensor([True, False, False]))
+    except ARTTestException as e:
+        art_warning(e)
+
+
+@pytest.mark.only_with_platform("pytorch")
+@pytest.mark.parametrize("ablation", ["block", "column"])
+def test_end_to_end_equivalence(art_warning, fix_get_mnist_data, fix_get_cifar10_data, ablation):
```

Comment:
Please replace this test with expected values to avoid relying on cloning from GitHub.

Reply:
updated with values to avoid a git clone.